packages/cerebral/src/Controller.js

```diff
@@ -166,7 +166,30 @@ class Controller extends FunctionTree {
       }, {})
     }
 
-    this.runTree(name, signal, payload)
+    this.runTree(name, signal, payload, (error) => {
+      if (error) {
+        const signalPath = error.execution.name.split('.')
+        const signalCatch = signalPath.reduce((currentModule, key, index) => {
+          if (index === signalPath.length - 1) {
+            return currentModule.signals[key].catch
+          }
+
+          return currentModule ? currentModule.modules[key] : undefined
+        }, this.module)
+
+        if (!signalCatch) {
+          throw error
+        }
+
+        if (Array.isArray(signalCatch)) {
+          this.runSignal('catch', signalCatch, error.payload)
+        } else if (error.name in signalCatch) {
```

Comment:
Not better check for isInstance instead of error.name?
This would allow to use groups of errors.
You could have e.g. a HTTPError class and some subclasses depending on the HTTP status code (ForbiddenError, NotFoundError, ConflictError,  UnprocessableEntityError etc.).
If you want to treat all HTTP errors the same you can use HTTPError which includes all subclasses. Or you can handle some subclasses individually. 
Not sure if this works in Javascript ;-).